Mongo_Adapter/AdapterActions/Push.cs

```diff
@@ -36,20 +38,19 @@ public partial class MongoAdapter
         /**** Public Methods                            ****/
         /***************************************************/
 
-        public override List<IObject> Push(IEnumerable<IObject> objects, string tag = "", Dictionary<string, object> config = null)
+        public override List<object> Push(IEnumerable<object> objects, string tag = "", PushType pushType = PushType.AdapterDefault, ActionConfig actionConfig = null)
         {
             // Check that the link is still alive
             if (m_Client.Cluster.Description.State == MongoDB.Driver.Core.Clusters.ClusterState.Disconnected)
-                return new List<IObject>();
+                return new List<object>();
 
             // Get the config
             bool replace = true;
-            if (config != null)
+            MongoConfig mongoConfig = actionConfig as MongoConfig;
+            if (actionConfig != null)
             {
-                if (config.ContainsKey("Tag"))
-                    tag = config["Tag"] as string;
```

Comment:
@adecler Was this actually needed? Why can't this work with the adapter input `tag`?
As you can see I preserved this [here](https://github.com/BHoM/Mongo_Toolkit/pull/82/files#r354815637) and [here](https://github.com/BHoM/Mongo_Toolkit/pull/82/files#r354815882), but I wonder if we couldn't remove it altogether.